### ISSUE TITLE
Working with isaacsim's python environment

### ISSUE THREAD

**Issue by johannespitz** [isaac-sim/IsaacAutomator#6]
I don't fully understand the container/cloud setup, so maybe there is a very different solution to the problem I have but ultimately I would like to work with the python environment that comes with isaacsim.
When I install the application via the omniverse launcher I just have to `alias isaacpython ~/.local/share/ov/pkg/isaac.../python.sh` and then work with `isaacpython` instead of `python`.

Now I found this repository, which automates the (very tedious) installation process and I would like to work with these images in the future. So is there a way to work with the `python.sh` here as well?
`/home/ubuntu/isaacsim.sh --cmd="./python.sh"` opens the interpreter I want, but so far I couldn't figure out how I can install pip packages into this environment.

**Comment by myurasov-nv** [comment]
@johannespitz thanks for the feedback!

You can install new pip packages inside Isaac Sim container with the following command: `/isaac-sim/python.sh -m pip install <module>`.

If you want packages installed on container startup, you can do `~/isaacsim.sh --cmd="./python.sh -m pip install <module>;  bash"`. This command will install a pip package and give you bash inside the container.

**Comment by myurasov-nv** [comment]
Please let me know if this solution works for you.

**Comment by johannespitz** [comment]
Thanks for the help.
I need to install quite a few packages and in particular and editable version of my own project.
Is there a the Dockerfile where I could make that part of the image such that I don't have to install packages for each run?

**Comment by myurasov-nv** [comment]
What you can do – place your custom `isaacsim-custom.dockerfile` in the `uploads/` dir and then build it automatically on a deployed system from the [`autorun.sh`](https://github.com/isaac-sim/IsaacAutomator?tab=readme-ov-file#autorun-script) script, giving it `isaacsim` tag, like so:

```sh
# contents of the autorun.sh
docker build -t isaacsim -f ~/uploads/isaacsim-custom.dockerfile .
# the rest of autorun.sh
```

Then you will have custom `isaacsim` container built on startup.

Your `isaacsim-custom.dockerfile` can look something like:

```Dockerfile
FROM nvcr.io/nvidia/isaac-sim:4.0.0
RUN /isaac-sim/python.sh -m pip install <modules>
```

**Comment by myurasov-nv** [comment]
@Ac31415 That dockerfile would be for building custom Isaac Sim, but nothing stops you from adding more lines to `uploads/autorun.sh`. Can you please describe in more details what needs to be installed? I'm thinking of adding ROS2 out of the box...

**Comment by Ac31415** [comment]
> [@Ac31415](https://github.com/Ac31415) That dockerfile would be for building custom Isaac Sim, but nothing stops you from adding more lines to `uploads/autorun.sh`. Can you please describe in more details what needs to be installed? I'm thinking of adding ROS2 out of the box...

Right after deploying to Azure, I followed what was said [here](https://docs.isaacsim.omniverse.nvidia.com/4.5.0/installation/install_ros.html) and [here](https://docs.ros.org/en/humble/Installation/Ubuntu-Install-Debs.html) to install ROS2 natively on that deployed instance. But then I got exactly the same error as in #32. And I also did the sourcing mentioned in the above documentations. So any suggestions on how to resolve that?
